catalog/resource_connection.go

```diff
@@ -100,13 +100,21 @@ func ResourceConnection() common.Resource {
 			if err != nil {
 				return err
 			}
+			if conn.Options == nil {
+				conn.Options = map[string]string{}
+			}
+			// remove not necessary parameters for builtin HMS
+			if val, exists := conn.Options["builtin"]; exists && val == "true" {
```

Comment:
Where is this `builtin` field documented? I don't see it on the linked website.

Reply:
yeah, that's a major problem with API spec - all these fields are buried under the docs for specific connection type.  For built-in hive metastore it's: https://docs.databricks.com/en/data-governance/unity-catalog/hms-federation/hms-federation-internal.html#language-SQL

Frankly speaking, I would prefer to have something like it's done for model serving and external models - each model type has its own structures with well defined what is computed/required/optional

Reply:
should we add it to the doc as well? a general complaint is that customers don't know which `option` to provide